blockly-core/core/ui/contract_editor/contract_editor.js

```diff
@@ -366,6 +367,40 @@ Blockly.ContractEditor.prototype.addExampleBlockFromMainBlockSpace = function(ex
 };
 
 /**
+ * Updates examples to allow any type of input.
+ * Used to avoid disconnection in the middle of a type switch-over.
+ * @private
+ */
+Blockly.ContractEditor.prototype.updateExamplesToAnyType_ = function () {
+  this.updateExampleInputsToType_(Blockly.BlockValueType.NONE);
+};
+
+/**
+ * Update examples' input types to match the function definition block's type
+ * @private
+ */
+Blockly.ContractEditor.prototype.updateExampleInputTypes_ = function () {
+  this.updateExampleInputsToType_(this.currentFunctionDefinitionType_());
+};
+
+Blockly.ContractEditor.prototype.updateExampleInputsToType_ = function (newType) {
+  var blocksToUpdate = this.exampleBlocks.concat(this.functionDefinitionBlock);
+  blocksToUpdate.forEach(function (exampleBlock) {
+    exampleBlock.updateUsageType(newType);
+  }, this);
+};
+
+/**
+ * Gets the current function definition type, assuming there is only one type
+ * check on the function definition block.
+ * @returns {Blockly.BlockValueType}
+ * @private
+ */
+Blockly.ContractEditor.prototype.currentFunctionDefinitionType_ = function () {
+  return this.functionDefinitionBlock.previousConnection.check_[0];
```

Comment:
Should we actually create a getter function on Connection? This seems to violate the rule of accessing someone else's private.